flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/sink/ConstraintEnforcer.java

```diff
@@ -172,36 +204,47 @@ public ConstraintEnforcer build() {
                 return new ConstraintEnforcer(
                         notNullEnforcer,
                         notNullFieldIndices,
-                        charLengthEnforcer,
+                        typeLengthEnforcer,
                         charFieldInfo != null
                                 ? charFieldInfo.stream().mapToInt(cfi -> cfi.fieldIdx).toArray()
                                 : null,
                         charFieldInfo != null
                                 ? charFieldInfo.stream().mapToInt(cfi -> cfi.length).toArray()
                                 : null,
-                        charFieldInfo != null ? buildShouldPad(charFieldInfo) : null,
+                        charFieldInfo != null ? buildCouldPad(charFieldInfo) : null,
+                        binaryFieldInfo != null
+                                ? binaryFieldInfo.stream().mapToInt(cfi -> cfi.fieldIdx).toArray()
```

Comment:
`cfi`?

Reply:
left over, now it's `fi` (FieldInfo), thx